libraries/api-documenter/src/test/MarkdownPageRenderer.test.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+/// <reference types='mocha' />
+
+// import { assert } from 'chai';
+import * as path from 'path';
+import { DiffTest } from '@microsoft/node-core-library';
+
+import { MarkdownPageRenderer } from '../MarkdownPageRenderer';
+import { IDomPage } from '../SimpleDom';
+import { Domifier } from '../Domifier';
+
+describe('MarkdownPageRenderer', () => {
+  it('renders markdown', (done: MochaDone) => {
+
+    const diffTest: DiffTest = new DiffTest();
+
+    const outputFolder: string = diffTest.getFolderPath(__dirname, 'MarkdownPageRenderer');
+
+    const renderer: MarkdownPageRenderer = new MarkdownPageRenderer(outputFolder);
+    const domPage: IDomPage = Domifier.createPage('Test page', 'test-id');
+    const outputFilename: string = renderer.writePage(domPage);
```

Comment:
When do you clean up the temp output files? Ideally we should not really write to the disk and only compare in memory.

Reply:
About cleaning up the output files:  Good feedback.  I have renamed getFolderPath() --> prepareFolder() which always deletes the entire folder contents.


About in-memory comparisons, we discussed that at the API review.  It is a good idea, but a more complicated implementation:

- The caller must provide a buffer or write stream.
- The files do still need to get written to disk when a diff is encountered
- Sometimes a developer wants to see the output files anyway; maybe we need a switch for that.

Let's introduce it later.

---
In reply to: [134927460](https://github.com/Microsoft/web-build-tools/pull/330#discussion_r134927460) [](ancestors = 134927460)